modules/pricer/src/main/java/com/opengamma/strata/pricer/credit/cds/CreditCurveZeroRateSensitivity.java

```diff
@@ -0,0 +1,609 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.credit.cds;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.function.DoubleUnaryOperator;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.google.common.collect.ComparisonChain;
+import com.opengamma.strata.basics.StandardId;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.currency.FxRateProvider;
+import com.opengamma.strata.market.sensitivity.MutablePointSensitivities;
+import com.opengamma.strata.market.sensitivity.PointSensitivity;
+import com.opengamma.strata.market.sensitivity.PointSensitivityBuilder;
+import com.opengamma.strata.pricer.ZeroRateSensitivity;
+
+/**
+ * Point sensitivity to the zero hazard rate curve.
+ * <p>
+ * Holds the sensitivity to the zero hazard rate curve at a specific date.
+ */
+@BeanDefinition(builderScope = "private")
+public final class CreditCurveZeroRateSensitivity
+    implements PointSensitivity, PointSensitivityBuilder, ImmutableBean, Serializable {
+
```

Comment:
Would it make more sense to store the legalEntityId and an actual `ZeroRateSensitivity object. That might reduce the duplication.